plugin/src/main/kotlin/com/ritense/plugin/service/PluginService.kt

```diff
@@ -113,15 +113,22 @@ class PluginService(
     fun deployPluginConfigurations(deploymentDto: PluginAutoDeploymentDto) {
         val plugin: PluginConfiguration
         val pluginDefinition = pluginDefinitionRepository.getById(deploymentDto.pluginDefinitionKey)
+        if (deploymentDto.id != null && pluginConfigurationRepository.existsById(
+                PluginConfigurationId.existingId(deploymentDto.id)
+            )
+        ) {
+            deletePluginConfiguration(PluginConfigurationId.existingId(deploymentDto.id))
+        }
         validateProperties(deploymentDto.properties!!, pluginDefinition)
-        plugin = pluginConfigurationRepository.save(
+        val pluginId = pluginConfigurationRepository.saveAndFlush(
```

Comment:
Since all plugin events are ran afterwards, if one of those calls fails it should really just not save the plugin configuration to the database. I think doing a save and flush here just introduces this as a new issue.

Reply:
When a plugin-event throws an error, the plugin-configuration is deleted

Reply:
So, first we delete an existing plugin configuration. Then we run the events. When the events fail, we are left without a pluginconfiguration where we used to have one.

Reply:
Not sure what you're suggesting here.

Most of the change are made for OpenNotifications. Because OpenNotifications tries to create an abonnement on the plugin-create-event and delete de abonnement on a plugin-delete-event. Which means that the plugin-events are a mandatory part every time the plugin-configuration is updated. Which wasn't done properly before.

However, the `saveAndFlush` is there for a different reason. The secret of a plugin-configuration is only encrypted when the entity is persisted. Which basically solved [this](https://ritensebv.slack.com/archives/G01Q25SVBHN/p1684755851723469) error that we were seeing on every startup

Reply:
At line 120 we delete a plugin configuration with that ID already exists in the database. Then, a new plugin configuration is created, and a flush occurs. This means the previous plugin configuration is now removed, and the new one has been saved. If then an even fails, this new plugin configuration is deleted from the database. This means now no plugin configuration with that ID can be found in the database, which is not what you would expect if a deployment fails.

Reply:
Flushing should not affect transaction management at all. When transaction is rolled back, all changes made to the database state (i.e. changes flushed by Hibernate) inside that transaction are rolled back.

Reply:
I agree that this peace of code isn't particularly beautiful. But this is the only way I could find to have the secret encryption and decryption happen in the correct order. Without making a breaking change, that is.